src/components/themed/ModalParts.tsx

```diff
@@ -27,13 +27,13 @@ export function ModalTitle(props: ModalTitleProps) {
   )
 }
 
-export function ModalMessage(props: { children: React.ReactNode; paddingRem?: number[] | number; isWarning?: boolean }) {
+export function ModalMessage(props: { children: React.ReactNode | string; paddingRem?: number[] | number; isWarning?: boolean }) {
   const { children, isWarning, paddingRem } = props
   const theme = useTheme()
   const styles = getStyles(theme)
   const padding = sidesToPadding(mapSides(fixSides(paddingRem, 0), theme.rem))
 
-  return <Text style={[styles.messageText, padding, isWarning && styles.warningText]}>{children}</Text>
+  return typeof children === 'string' ? <Text style={[styles.messageText, padding, isWarning && styles.warningText]}>{children}</Text> : <>{children}</>
 }
```

Comment:
I think this is the wrong layering. The `TextInputModal` component should be should be doing this instead, like `typeof message === 'string' ? <ModalMessage>{message}</ModalMessage> : message`. That way the `ModalMessage` component retains its original purpose, which is to display text.

Next, instead of passing custom CSS to `EdgeText`, you would just use `ModalMessage` and `ModalTitle` as normal, like:

```tsx
<TextInputModal
  messge={
    <ScrollView>
      <ModalTitle>{...}</ModalTitle>
      <ModalMessage>{...}</ModalMessage>
    </ScrollView>
  }
/>
```

Reply:
I have tested this change locally, and using the `ModalTitle` and `ModalMessage` inside a scroll view does look better, and requires no additional CSS.